server/athenian/api/internal/miners/jira/issue.py

```diff
@@ -596,6 +596,16 @@ async def _fill_issues_with_mapped_prs_info(
         f"Leading({NodePullRequestJiraIssues.__tablename__} *VALUES* "
         f"{NodePullRequest.__tablename__})",
     )
+    if len(issues.index.values) > 300:
+        # with many issues sometimes our Leading hint confuses planner,
+        # a sequential scan is done on NodePullRequestJiraIssues
+        # forcing Parallel (which would be probably done anyway with a good plan) avoids this
+        for hint in (
+            f"HashJoin({NodePullRequestJiraIssues.__tablename__} *VALUES)",
```

Comment:
Note: this can also happen if we've got more than one meta id. In that case, should use a UNION

Reply:
We already use the union for multiple meta ids, or do you mean that we should always have those hints if we using a union ?